pySDC/core/Sweeper.py

```diff
@@ -270,40 +280,102 @@ def rho(x):
             self.parallelizable = True
 
         elif qd_type == "MIN-SR-S":
+
             M = QDmat.shape[0] - 1
-            Q = coll.Qmat[1:, 1:]
-            nodes = coll.nodes
+            quadType = coll.quad_type
+            nodeType = coll.node_type
 
-            nCoeffs = M
-            if coll.quad_type in ['LOBATTO', 'RADAU-LEFT']:
-                nCoeffs -= 1
-                Q = Q[1:, 1:]
-                nodes = nodes[1:]
+            # Main function to compute coefficients
+            def computeCoeffs(m, a=None, b=None):
+                coll = CollBase(num_nodes=m, node_type=nodeType, quad_type=quadType)
 
-            def func(coeffs):
-                coeffs = np.asarray(coeffs)
-                kMats = [(1 - z) * np.eye(nCoeffs) + z * np.diag(1 / coeffs) @ Q for z in nodes]
-                vals = [np.linalg.det(K) - 1 for K in kMats]
-                return np.array(vals)
+                Q = coll.Qmat[1:, 1:]
+                nodes = coll.nodes
 
-            coeffs = sp.optimize.fsolve(func, nodes / M, xtol=1e-13)
+                nCoeffs = len(nodes)
+                if quadType in ['LOBATTO', 'RADAU-LEFT']:
+                    nCoeffs -= 1
+                    Q = Q[1:, 1:]
+                    nodes = nodes[1:]
 
-            if coll.quad_type in ['LOBATTO', 'RADAU-LEFT']:
-                coeffs = [0] + list(coeffs)
+                if nCoeffs == 1:
+                    coeffs = np.diag(Q)
+
+                else:
+
+                    def nilpotency(coeffs):
+                        """Function verifying the nilpotency from given coefficients"""
+                        coeffs = np.asarray(coeffs)
+                        kMats = [(1 - z) * np.eye(nCoeffs) + z * np.diag(1 / coeffs) @ Q for z in nodes]
+                        vals = [np.linalg.det(K) - 1 for K in kMats]
+                        return np.array(vals)
+
+                    if a is None:
+                        coeffs0 = nodes / m
+                    else:
+                        coeffs0 = a * nodes**b / m
+
+                    with warnings.catch_warnings():
+                        warnings.simplefilter("ignore")
+                        coeffs = sp.optimize.fsolve(nilpotency, coeffs0, xtol=1e-15)
+
+                # Handle first node equal to zero
+                if quadType in ['LOBATTO', 'RADAU-LEFT']:
+                    coeffs = np.asarray([0.0] + list(coeffs))
+                    nodes = coll.nodes
+
+                return coeffs, nodes
+
+            def fit(coeffs, nodes):
+                """Function fitting given coefficients to a power law"""
+
+                def law(ab):
+                    a, b = ab
+                    return np.linalg.norm(a * nodes**b - coeffs)
+
+                sol = sp.optimize.minimize(law, [1.0, 1.0], method="nelder-mead")
+                return sol.x
+
+            # TODO : put this somewhere as a parameter ...
```

Comment:
So?

Reply:
Removed it (eventually put it back later, but that's just for experiments)